src/app/application.cpp

```diff
@@ -149,8 +149,13 @@ Application::Application(int &argc, char **argv)
     const QString profileDir = portableModeEnabled
         ? QDir(QCoreApplication::applicationDirPath()).absoluteFilePath(DEFAULT_PORTABLE_MODE_PROFILE_DIR)
         : m_commandLineArgs.profileDir;
-    const QString appId = QLatin1String("qBittorrent-") + Utils::Misc::getUserIDString() + '@' + profileDir
-            + (m_commandLineArgs.configurationName.isEmpty() ? QString {} : ('/' + m_commandLineArgs.configurationName));
+    const QString instanceId = (profileDir + (m_commandLineArgs.configurationName.isEmpty() ? QString {} : ('/' + m_commandLineArgs.configurationName)))
+#ifdef Q_OS_WIN
+        .toLower();
```

Comment:
IMO you can make `instanceId` non-const and reassign it. Or if you don't mind having duplicates:
```c++
#ifdef Q_OS_WIN
    const QString instanceId = ....toLower();
#else
    const QString instanceId = ...;
#endif
```

Reply:
>you can make instanceId non-const and reassign it. 

I'm surprised to hear you say that...

>Or if you don't mind having duplicates:

Current notation clearly shows the difference between implementation on different OSes. Why don't you like it?
If you don't insist on its changing I would leave it as is.

Reply:
> > you can make instanceId non-const and reassign it.
> 
> I'm surprised to hear you say that...

Because I think even making it non-const is better than the current state.

>Why don't you like it?

Personally I think cutting off an expression is bad for maintainability, such as what if we need to add more operations to it?

Reply:
Well, I'll change it.

Reply:
Changed.